### PR TITLE
#18 fix list in ticket

### PR DESCRIPTION
**Pull request by curtiswellborn**
1. add .netcore 2.1 support
2. bump .netstandard from 1.3 to 2.0
3. deadlocks
4. ReadFromFileAsync
5. write unit tests
6. bump version number

### CODE REVIEW

File: FeedReader/FeedReader.cs [hunk 1]
```diff
@@ -190,6 +198,30 @@ public static Feed ReadFromFile(string filePath)
...
 
+        /// <summary>
+        /// reads a feed from a file
+        /// </summary>
+        /// <param name="filePath">the path to the feed file</param>
+        /// <returns>parsed feed</returns>
+        public static async Task<Feed> ReadFromFileAsync(string filePath)
+        {
+            byte[] result;
+            #if NETCOREAPP2_1
```
Comment (curtiswellborn) [inline FeedReader/FeedReader.cs]: preprocessor for .net core 2.1

File: FeedReader/FeedReader.csproj [hunk 2]
```diff
@@ -1,31 +1,31 @@
...
     <Company>Reiter ITS</Company>
     <Product>CodeHollow.FeedReader</Product>
-    <Description>FeedReader is a .net standard 1.3 library used for reading and parsing RSS and ATOM feeds. Supports RSS 0.91, 0.92, 1.0, 2.0 and ATOM. Developed because existing libraries do not work with different languages, encodings or have other issues. Library tested with multiple languages and feeds.</Description>
+    <Description>FeedReader is a .net standard 2.0 library used for reading and parsing RSS and ATOM feeds. Supports RSS 0.91, 0.92, 1.0, 2.0 and ATOM. Developed because existing libraries do not work with different languages, encodings or have other issues. Library tested with multiple languages and feeds.</Description>
 	<Copyright>Copyright 2018</Copyright>
     <PackageLicenseUrl>https://github.com/codehollow/FeedReader/blob/master/LICENSE</PackageLicenseUrl>
     <PackageProjectUrl>https://github.com/codehollow/FeedReader/</PackageProjectUrl>
     <RepositoryUrl>https://github.com/codehollow/FeedReader/</RepositoryUrl>
-    <Version>1.1.4</Version>
+    <Version>1.1.5</Version>
```
Comment (curtiswellborn) [inline FeedReader/FeedReader.csproj]: Was not sure if you bump versions for this project...I can remove this if need be

File: FeedReader/FeedReader.csproj [hunk 3]
```diff
@@ -1,31 +1,31 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>netstandard1.3;net452</TargetFrameworks>
+    <TargetFrameworks>netstandard2.0;net452;netcoreapp2.1</TargetFrameworks>
```
Comment (curtiswellborn) [inline FeedReader/FeedReader.csproj]: Not sure if yall want this bumped to 2.0 or not. Let me know and Ill move it back down(it should be backwards compatible so it should matter). I just needed core 2.1

**Comment by codehollow** [comment]
thanks @curtiswellborn: I already uploaded the updated nuget package!

**Pull request merged**